canvas/biplot.py

```diff
@@ -161,150 +213,115 @@ def make_biplot(samples,
                 marker=samp_marker, linestyle='',
                 ms=samp_ms, alpha=samp_alpha)
 
-    if (sample_metadata is not None and sample_color_dict is None):
-        sample_groups = samples.groupby(sample_metadata[sample_color_category])
-
-        if len(sample_drop_list) > 0:
+    if sample_metadata is not None:
 
-            def dropf(x):
-                return x not in sample_drop_list
-            index_drop = sample_metadata[sample_color_category].apply(dropf)
+        if sample_color is not None:
+            if (type(sample_color) is dict or
+                    type(sample_color) is OrderedDict):
+                sample_color = pd.Series(sample_color.values(),
+                                         index=sample_color.keys())
 
-            samp_r = samples.loc[sample_metadata.index]
-            samp_met_r = sample_metadata.loc[index_drop][sample_color_category]
+        if len(samp_drop_list) > 0:
```

Comment:
I'm not so sure if we should have this.  What exactly is this used for?  And can we move this preprocessing outside of this function?